bin/strata-client/src/args.rs

```diff
@@ -1,171 +1,222 @@
+            overridden = true
+        }
+        if let Some(rpc_host) = &self.rpc_host {
+            config.client.rpc_host = rpc_host.to_string();
+            overridden = true
+        }
+        if let Some(rpc_port) = &self.rpc_port {
+            config.client.rpc_port = *rpc_port;
+            overridden = true
+        }
+        Ok(overridden)
+    }
 
-    /// Timeout duration for btc request retries in ms. Defaults to `1000`.
-    #[argh(
-        option,
-        description = "max interval between bitcoin RPC retries in ms (default: 1000)"
-    )]
-    pub bitcoind_retry_interval: Option<u64>,
+    /// Override config using the generic overrides. The idea here is to first convert the
+    /// [`Config`] to json and then apply the overrides by splitting the key by a dot.
+    fn override_generic(&self, config: &mut Config) -> anyhow::Result<bool> {
+        let original = config.clone();
+        // Convert config as json
+        let mut json_config = to_value(&mut *config).expect("Config json serialization failed");
 
-    #[argh(option, short = 'n', description = "L1 network to run on")]
-    pub network: Option<Network>,
+        for (path, val) in parse_overrides(&self.overrides)?.iter() {
+            apply_override(path, val, &mut json_config)?;
+        }
+        // Convert back to json.
+        *config =
+            from_value(json_config).expect("Should be able to create Config from serde json Value");
+        Ok(original != *config)
+    }
+}
 
-    #[argh(switch, description = "start sequencer bookkeeping tasks")]
-    pub sequencer: bool,
+type Override = (Vec<String>, String);
+
+/// Parse valid overrides. This first splits the entries by '=' to get key and value and then splits
+/// the key by '.' which is the update path.
+fn parse_overrides(overrides: &[String]) -> anyhow::Result<Vec<Override>> {
+    let mut result = Vec::new();
+    for item in overrides {
+        let (key, value) = item
+            .split_once("=")
+            .ok_or(anyhow!("Invalid override: must be in 'key=value' format"))?;
+        let path: Vec<_> = key.split(".").map(|x| x.to_string()).collect();
+        result.push((path, value.to_string()));
+    }
+    Ok(result)
+}
 
-    #[argh(option, description = "sequencer rpc host:port")]
-    pub sequencer_rpc: Option<String>,
+/// Apply override to config.
+fn apply_override(path: &[String], str_value: &str, config: &mut Value) -> anyhow::Result<()> {
+    match path {
+        [key] => {
+            config[key] = parse_value(str_value)?;
+        }
+        [key, other @ ..] => {
+            apply_override(other, str_value, &mut config[key])?;
+        }
+        [] => return Err(anyhow!("Invalid override path")),
+    };
+    Ok(())
+}
 
-    #[argh(option, description = "reth authrpc host:port")]
-    pub reth_authrpc: Option<String>,
+/// Parses a string `T`. If parsing fails, attempts to parse it again by converting it to JSON
```

Comment:
```suggestion
/// Parses a string into a `T`. If parsing fails, attempts to parse it again by converting it to JSON
```

The `T` is the type that you're parsing the string into.